checker/jdk/nullness/src/javax/xml/parsers/DocumentBuilder.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2000, 2006, Oracle and/or its affiliates. All rights reserved.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ */
+
+package javax.xml.parsers;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.xml.validation.Schema;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.DOMImplementation;
+
+import org.xml.sax.EntityResolver;
+import org.xml.sax.ErrorHandler;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import org.checkerframework.checker.nullness.qual.* ;
+
+/**
+ * Defines the API to obtain DOM Document instances from an XML
+ * document. Using this class, an application programmer can obtain a
+ * {@link Document} from XML.<p>
+ *
+ * An instance of this class can be obtained from the
+ * {@link DocumentBuilderFactory#newDocumentBuilder()} method. Once
+ * an instance of this class is obtained, XML can be parsed from a
+ * variety of input sources. These input sources are InputStreams,
+ * Files, URLs, and SAX InputSources.<p>
+ *
+ * Note that this class reuses several classes from the SAX API. This
+ * does not require that the implementor of the underlying DOM
+ * implementation use a SAX parser to parse XML document into a
+ * <code>Document</code>. It merely requires that the implementation
+ * communicate with the application using these existing APIs.
+ *
+ * @author <a href="mailto:Jeff.Suttor@Sun.com">Jeff Suttor</a>
+ */
+
+public abstract class DocumentBuilder {
+
+
+    /** Protected constructor */
+    protected DocumentBuilder () {
+    }
+
+    /**
+     * <p>Reset this <code>DocumentBuilder</code> to its original configuration.</p>
+     *
+     * <p><code>DocumentBuilder</code> is reset to the same state as when it was created with
+     * {@link DocumentBuilderFactory#newDocumentBuilder()}.
+     * <code>reset()</code> is designed to allow the reuse of existing <code>DocumentBuilder</code>s
+     * thus saving resources associated with the creation of new <code>DocumentBuilder</code>s.</p>
+     *
+     * <p>The reset <code>DocumentBuilder</code> is not guaranteed to have the same {@link EntityResolver} or {@link ErrorHandler}
+     * <code>Object</code>s, e.g. {@link Object#equals(Object obj)}.  It is guaranteed to have a functionally equal
+     * <code>EntityResolver</code> and <code>ErrorHandler</code>.</p>
+     *
+     * @throws UnsupportedOperationException When implementation does not
+     *   override this method.
+     *
+     * @since 1.5
+     */
+    @SuppressWarnings("nullness")
```

Comment:
For every `@SuppressWarnings` annotation, please add a brief comment on the same line explaining why the warning is being suppressed.  That is, please justify that the code is safe, even though the Nullness Checker is unable to prove that fact.  For some examples, see https://checkerframework.org/manual/#suppresswarnings-best-practices .